components/post_view/post_attachment.jsx

```diff
@@ -297,9 +267,25 @@ export default class PostAttachment extends React.PureComponent {
 
         let text;
         if (data.text) {
+            const shouldCollapse = this.shouldCollapse();
+            const collapsed = shouldCollapse && this.state.collapsed;
+            const textHTML = collapsed ? this.getCollapsedTextHTML() : TextFormatting.formatText(this.props.attachment.text || '');
+            const collapseMessage = collapsed ? localizeMessage('post_attachment.more', 'Show more...') : localizeMessage('post_attachment.collapse', 'Show less...');
+
             text = (
                 <div className='attachment__text'>
-                    {messageHtmlToComponent(this.state.textHTML, false)}
+                    {messageHtmlToComponent(textHTML, false)}
+                    {shouldCollapse &&
+                        <div>
+                            <a
+                                className='attachment-link-more'
+                                href='#'
+                                onClick={this.toggleCollapseState}
+                            >
+                                {collapseMessage}
+                            </a>
+                        </div>
+                    }
                 </div>
             );
         }
```

Comment:
Please add component test so we have coverage to `PostAttachment`.

Reply:
*cries*

Reply:
do I really need to?

Reply:
Yes, usually for component migration from redux and/or to pure component, or refactor initiative like this.

Reply:
@koxen Let me know if you need help on test. Thanks!

Reply:
@saturninoabril I would appreciate your help very much because I never worked with this kind of tests :)!

Reply:
Sure no problem. Please help rebase this to latest, then we're good to merge.
I'll submit PR for test separately (will put link to this).

Reply:
Awesome! Thanks for making an exception this time! :)